www/search.py

```diff
@@ -105,18 +144,54 @@ def GetResult(request):
 	LOWER(c.name) as category,
 	f.size as size,
 	f.timestamp as timestamp,
+	GROUP_CONCAT(fk.keyword ORDER BY keyword ASC SEPARATOR ',') as keywords, 
 	f.metadata as metadata
 	FROM file as f
 	LEFT JOIN categories as c ON f.cid=c.cid
 	LEFT JOIN tag as t ON  f.fid=t.fid
-	WHERE c.cid>0
-	AND f.status=1
-	%s
-	ORDER BY f.timestamp DESC
+	LEFT JOIN file_keyword AS fk ON (f.fid=fk.fid)
+	"""
+	
+	keywordsSet = False
+	if "keywords" in request and request["keywords"] != "":
+		keywordsSet = True
+
+	# use different query to show only latest version for each name (maps only, for now)
+	if "latestOnly" in request and int(request["latestOnly"]) == 1 and "category" in request:
```

Comment:
not sure how this should work: getting the latest version number isn't just a string compare: 
i.e. "10" < "9" but 10 > 9

sort at least by id, maybe by timestamp.

how does this get the "latest" version?

Reply:
that's only part of the query, the order by is still added in the end.

Indeed, it's not perfect, but works for 99% of cases from what i've seen. 

extracting the version and converting to integer would imply changing where f.name=f2.latest as well and might be much slower.

Reply:
A fix is storing the converted integer from the version string as an extra "version_number" column and use that to figure out the latest for each map.

Reply:
i did this on the latest commit, named it version_sort_number which uses a floating point number